src/recovery/cubic.rs

```diff
@@ -646,3 +671,22 @@ mod tests {
         assert_eq!(r.cwnd(), cwnd_prev + r.max_datagram_size);
     }
 }
+
+fn checkpoint(r: &mut Recovery) {
```

Comment:
I think we just need to have a stub here and let each PR has actual implementation. Same to rollback()

Reply:
Why? This is the generic part from draft-ietf-tcpm-rfc8312bis, why would each PR re-implement the same thing?